File: Linked List/Linked Queue/Implementation in Java/LinkedQueue.java
import mypackage;

// The LinkedQueue Implementation
public class LinkedQueue {
	int size;          // the counter of LinkedQueue
	Node rear;
	Node front;
    
    // LinkedQueue constractor
	public LinkedQueue() {
		this.size = 0;
		this.rear = null;
		this.front = null;
	}

	// Adding element in the LinkedQueue
	public void enqueue(int element) {
		Node temp = new Node(element);

		if(rear == null) {   // if the list is empty make the rear and front node copy temp
			rear = front = temp;
			return;
		}

		rear.next = temp;       // assign the next to the temp 
		rear = temp;        //  assign the temp to the top
		size++;          // counter to increase the size
	}

	// Removing  first element from the top
	public int dequeue() {
	    // condation to check if the queue is empty or not
		if(front == null)
			return -1;

		int dequeued = front.data; 
		front = front.next;   // assign the next of the front into front to make the pointer look at the next
		size--;          // decrease the counter after removing the element

		// If Queue is Empty
		if (front == null)
			rear = null;

		return dequeued;
	}

	// peek function to return the last element without removing it
	public int peek() {
	    // condation to check if the queue is empty or not
		if(front == null)
			return -1;
		return this.front.data;     // return the data of the front
	}

	// size function to return the size of the stack
	public int size() {
	    // condation to check if the queue is empty or not
		if(front == null)
			return -1;
		return this.size;
	}

	// Prints the stack elements with indices
	public void print() {
	    // condation to check if the queue is empty or not
		if(front == null)
			return;
		Node temp = front;
		int i = 0;
		while(temp != null) {
			System.out.println("[" + (i++) + "]" + "->" + temp.data);
			temp = temp.next;
		}
	}

	// isEmpty function to check if stack is full or not
	public boolean isEmpty() {
	    // condation to check if the queue is empty or not
		if(front == null)
			return true;
		return false;
	}

	// isFull function to check if stack is full or not
	public boolean isFull() {
	    // condation to check if the queue is empty or not
		if(front == null)
			return false;
		return true;
	}

	// Clears all nodes in the stack
	public void clear() {
		this.size = 0;
		this.rear = null;
		this.front = null;
	}
}

File: Linked List/Linked Queue/Implementation in C/LinkedQueue.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "LinkedQueue.h"

// Function to create an empty queue
Queue* createQueue() {
	Queue* queue = (Queue*)malloc(sizeof(Queue));
	queue->front = queue->rear = NULL;
	return queue;
}

// push function to insert elements in the queue at the rear
void enqueue(Queue** queue, int* size, int element) {
	Node* temp = createNode(element);

	if((*queue)->rear == NULL) {
		(*queue)->rear = (*queue)->front = temp;
		return;
	}

	(*queue)->rear->next = temp;
	(*queue)->rear = temp;
	(*size)++;
}

// pop function to remove the front element
int dequeue(Queue* queue, int* size) {
	// If the queue is empty
	if (queue->front == NULL) {
		printf("Queue Underflow!\n");
		return -1;
	}

	Node* temp = queue->front;
	int dequeued = queue->front->data;
	queue->front = queue->front->next;
	(*size)--;

	if (queue->front == NULL) // If the queue becomes empty, update rear to NULL
		queue->rear = NULL;

	free(temp);

	return dequeued;
}

// peek function to return the front element without removing it
int peek(Queue* queue) {
	// condation to check if the queue is empty or not
	if(queue->front == NULL)
		return -1;
	return queue->front->data;
}

// Print the queue elements
void print(Queue* queue) {
	// condation to check if the queue is empty or not
	if(queue->front == NULL)
		return;
	int i = 0;
	Node* temp = queue->front;
	while(temp != NULL) {
		printf("[%d] -> %d\n", (i++), temp->data);
		temp = temp->next;
	}
}

// isEmpty function to check if queue is full or not
bool isEmpty(Queue* queue) {
	// Condtion to check if it is empty or not
	if(queue->front == NULL || queue == NULL)
		return true;
	return false;
}

// isFull function to check if Queue is full or not
bool isFull(Queue* queue) {
	// Condtion to check if it is full or not
	if(queue->front == NULL || queue == NULL)
		return false;
	return true;
}

// freeLinkedQueue function to free out the queue to avoid memory leakage
void freeLinkedQueue(Queue** queue, int* size) {
	if (queue == NULL || *queue == NULL)
		return;

	Node* current = (*queue)->front;
	while (current != NULL) {
		Node* temp = current;
		current = current->next;
		free(temp);
	}
	free(*queue);
	*queue = NULL;
	*size = 0;
}
